src/eepp/ui/doc/languages/vb.cpp

```diff
@@ -8,8 +8,7 @@ void addVisualBasic() {
 	SyntaxDefinitionManager::instance()->add(
 
 		{ "Visual Basic",
-		  { "%.bas$", "%.cls$", "%.ctl$", "%.dob$", "%.dsm$", "%.dsr$", "%.frm$", "%.pag$", "%.vb$",
-			"%.vba$", "%.vbs$" },
+		  { "%.vb$", "%.vbs$" },
```

Comment:
This change is incorrect. All the previous extensions are valid Visual Basic extensions, and yes, it overlaps with SmallBasic and any BASIC language.
When languages overlaps in extensions ecode will detect it and let the user decide the file type of that particular file (it will display a floating window at the upper-right of the editor, you can test it with any .v file that overlaps V lang and Verilog).
What needs to be done when this happens is to set one of the languages as the "primary" or "default selected" language, the decision is always made with a single simple rule: the most popular language should be the default. In this case Visual Basic is vastly more popular so it should be set as the primary language. To do this you just simply need to add `.setExtensionPriority( true );` at the language definition, so in this case line 79 in `vb.cpp`:
please replace:
`		} );`
with
`		} ).setExtensionPriority( true );`

And roll-back the extensions change.

Reply:
> This change is incorrect. All the previous extensions are valid Visual Basic extensions

I don't think so. I simply fix what I think is incorrect. How did you get this list of extensions for Visual Basic in the first place? I think this list of extensions for Visual Basic is wrong. For example, `.frm` is simply an abbreviation for `form`. Delphi Form also uses the extension `.frm`.

Reply:
I programmed in VB6 for more than 10 years. All those extensions are valid, GUI forms are in `.frm` extension, modules in `.bas`, classes in `.cls`, custom controls `.ctl`, etc. They are all pretty standard. A simple google search will give you the answer: [here](https://stackoverflow.com/a/37933465/429096). A more extended list where you will find some of the others listed is [here](https://stackoverflow.com/a/62175248/429096).